### ISSUE TITLE
don't use proxyfix if not using wsgi

### ISSUE THREAD

**Issue by lukasjuhrich** [agdsn/sipa#354]
The ProxyFix does not work with flask's direct HTTP server, as it assumes data in the `wsgi.*`-environment keys. This breaks setups not using `uwsgi`, but plain ProxyPass, such as the `sipa_debug` setup in the `development.yml` docker-compose config.

**Comment by sebschrader** [comment]
What is Flask's server assuming exactly?

**Comment by lukasjuhrich** [comment]
To be more specific, the stacktrace is:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1997, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.6/site-packages/werkzeug/contrib/fixers.py", line 152, in __call__
    return self.app(environ, start_response)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1540, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1512, in handle_user_exception
    return self.handle_http_exception(e)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1471, in handle_http_exception
    return handler(e)
  File "/home/sipa/sipa/sipa/blueprints/generic.py", line 63, in error_handler_redirection
    message=message
  File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/usr/local/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/home/sipa/sipa/sipa/templates/error.html", line 2, in top-level template code
    {% set page_title = _("Fehler {} - {}").format(errorcode, message) %}
  File "/home/sipa/sipa/sipa/templates/base.html", line 160, in top-level template code
    <a href="{{ url_self(locale=locale.language) }}">
  File "/home/sipa/sipa/sipa/utils/__init__.py", line 100, in url_self
    raise RuntimeError("No endpoint available.")
RuntimeError: No endpoint available.
```

Where the nginx config is as given in `example/nginx.conf`

```conf
http {
    server {
        listen 80;
        server_name localhost;

        proxy_set_header Host $http_host;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_redirect off;
        # […]
        location = /sipa_debug { rewrite ^ /sipa_debug/; }
        location /sipa_debug/ {
            proxy_pass http://sipa_debug:5000;
            proxy_set_header Host $host;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Scheme $scheme;
            proxy_set_header X-Script-Name /sipa_debug;
        }

    }
}

```

**Comment by sebschrader** [comment]
The fix of #355 in 9e3ec85c15b8ca109fe3056fba7065d8c04ac2e9 addresses the exception you reported.

Could you check, if changes to ProxyFix are still required?

**Comment by lukasjuhrich** [comment]
Currently, the url root is not recognized correctly, leaving requests not mapping correctly. I'm really confused about which part of the stack is actually responsible for parsing `SCRIPT_ROOT` (or `SCRIPT_NAME`? Is this even standardized?)

**Comment by lukasjuhrich** [comment]
Ah, found it: https://github.com/pallets/werkzeug/blob/1c70c3e1855d32a285843c0851f52e20160f6656/werkzeug/wrappers.py#L604

As it appears, `request.environ` does not contain `SCRIPT_NAME` at all (when trying to access `localhost/sipa_debug/news`)

**Comment by lukasjuhrich** [comment]
This has been effectively resolved by 2668b07719497225ac7b893f553916142671f0fe
